evm-bridging/src/BridgedTopShotMoments.sol

```diff
@@ -104,10 +120,90 @@ contract BridgedTopShotMoments is
         override(ERC721Upgradeable, ERC721EnumerableUpgradeable)
         returns (address)
     {
+        // Add the beforeTokenTransfer hook
+        _beforeTokenTransfer(_ownerOf(tokenId), to, tokenId);
+
+        // Call parent implementation
         return super._update(to, tokenId, auth);
     }
 
     function _increaseBalance(address account, uint128 value) internal override(ERC721Upgradeable, ERC721EnumerableUpgradeable) {
         super._increaseBalance(account, value);
     }
+
+    function setRoyaltyInfo(RoyaltyInfo calldata newInfo) external onlyOwner {
+        // Revert if the new royalty address is the zero address.
+        if (newInfo.royaltyAddress == address(0)) {
+            revert RoyaltyAddressCannotBeZeroAddress();
+        }
+
+        // Revert if the new basis points is greater than 10_000.
+        if (newInfo.royaltyBps > 10_000) {
+            revert InvalidRoyaltyBasisPoints(newInfo.royaltyBps);
+        }
+
+        // Set the new royalty info.
+        _royaltyInfo = newInfo;
+
+        // Emit an event with the updated params.
+        emit RoyaltyInfoUpdated(newInfo.royaltyAddress, newInfo.royaltyBps);
+    }
+
+    function royaltyAddress() external view returns (address) {
+        return _royaltyInfo.royaltyAddress;
+    }
+
+    function royaltyBasisPoints() external view returns (uint256) {
+        return _royaltyInfo.royaltyBps;
+    }
+
+    /**
+     * @dev Implements the IERC2981 interface.
+     */
+    function royaltyInfo(
+        uint256 /* _tokenId */,
+        uint256 _salePrice
+    ) external view returns (address receiver, uint256 royaltyAmount) {
```

Comment:
Does this mean the contract needs to implement the `IERC2981` interface?

Reply:
Added inheritance, thank you! (`IERC2981` also returned in supportsInterface)